### ISSUE TITLE
Redirect from google.com to www.google.com to https://www.google.com...

### ISSUE THREAD

**Issue by as-com** [alex/what-happens-when#88]
The question is "What happens when you type google.com into your browser and press enter?"

Google redirects google.com to www.google.com with a 301 redirect. Then it redirects to https://www.google.com/?gws_rd=ssl with a 302 redirect.

It's not mentioned in the text.

**Comment by CoolOppo** [comment]
> Then it redirects to https://www.google.com/?gws_rd=ssl with a 302 redirect.

Incorrect. Google is a preloaded HSTS host, so it will always load with HTTPS. HSTS is mentioned in the text, but it could probably use a little more explanation.

I agree with you that the redirection from `google.com` to `www.google.com` should be mentioned though.

**Comment by as-com** [comment]
I actually saw Google redirecting to ?gws_rd=ssl in the Network tab of the dev tools in Chrome (incognito mode). I don't know what happened. It went straight to https://... every time after that.

However, this will always happen in browsers that do not support HSTS.

**Comment by CoolOppo** [comment]
Yeah I guess I wasn't paying attention to the URL parameter.

**Comment by StevenBlack** [comment]
Take note: For requests originating in Canada, and presumably other countries, the ultimate address is a localized TLD.  In Canada that's

  `https://www.google.ca`

**Comment by sharmavino** [comment]
NO

**Comment by BenjaminProgram** [comment]
@sharmavino No, what?
